packages/dcos-integration-test/extra/test_iam.py

```diff
@@ -0,0 +1,109 @@
+"""
+A collection of tests covering basic IAM functionality in DC/OS.
+
+Assume that access control is activated in Master Admin Router (could be
+disabled with `oauth_enabled`) and therefore authenticate individual HTTP
+requests by using dcos_api_session.
+"""
+import json
+import logging
+import time
+
+import cryptography.hazmat.backends
+import jwt
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from jwt.utils import base64url_decode
+
+
+__maintainer__ = 'jgehrcke'
+__contact__ = 'security-team@mesosphere.io'
+
+
+log = logging.getLogger(__name__)
+
+
+def _generate_rsa_keypair():
+    """
+    Generate an RSA keypair. Serialize the public key in the the X.509
+    SubjectPublicKeyInfo/OpenSSL PEM public key format (RFC 5280). Serialize the
+    private key in the PKCS#8 (RFC 3447) format.
+
+    Returns:
+        (private key, public key) 2-tuple, both unicode
+        objects holding the serialized keys.
+    """
+    private_key = rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048,
+        backend=cryptography.hazmat.backends.default_backend())
+
+    privkey_pem = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption())
+
+    public_key = private_key.public_key()
+    pubkey_pem = public_key.public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.SubjectPublicKeyInfo)
+
+    return privkey_pem.decode('ascii'), pubkey_pem.decode('ascii')
+
+
+default_rsa_privkey, default_rsa_pubkey = _generate_rsa_keypair()
+
+
+def test_service_account_create_login_delete(
+        dcos_api_session, noauth_api_session):
+
+    # Create service user account, share the public key with the IAM.
+    serviceuid = 'testservice'
+    r = dcos_api_session.put(
+        '/acs/api/v1/users/' + serviceuid,
+        json={'description': 'foo', 'public_key': default_rsa_pubkey}
+    )
+    assert r.status_code == 201, r.text
+
```

Comment:
Should we have a `try...finally` starting here to ensure that the service account deletion below gets run, even if an error occurs in between?

Reply:
I would punt on that for now if that's OK. In the long run we need a reliable, proper IAM cleanup fixture I think.